spec/features/visitors/password_recovery_spec.rb

```diff
@@ -142,15 +143,14 @@
     before do
       @user = create(:user, :signed_up)
 
-      visit new_user_password_path
+      visit_idp_from_sp_with_ial1(:oidc)
+      click_link t('links.passwords.forgot')
       fill_in t('account.index.email'), with: @user.email
       click_button t('forms.buttons.continue')
 
-      raw_reset_token, db_confirmation_token =
-        Devise.token_generator.generate(User, :reset_password_token)
-      UpdateUser.new(user: @user, attributes: { reset_password_token: db_confirmation_token }).call
-
-      visit edit_user_password_path(reset_password_token: raw_reset_token)
+      Capybara.current_session.reset!
```

Comment:
simulate opening in a different browser with different cookies?
if so we have the `perform_in_browser` pattern? 

https://github.com/18F/identity-idp/blob/1df1fb6d0adf4516d568e7b80e0035429ea56b22/spec/features/users/sign_in_spec.rb#L381-L391

Reply:
Yeah the idea is just to have a fresh session at the point that the user clicks the link in the email, simulating that they'd tried from a different device / browser / after their original session expired. I'd considered some of these other helpers, but I worried it'd require a more extensive redo of how the tests are structured, since this setup and the initial click on the link email all happens in the `before` block.

Looking at it again, maybe I can at least add a wrapping `perform_in_browser` for the steps above which initiate the password reset, and then proceed from the "main" browser afterward. I'll give it a shot.

Reply:
Updated in 6a1f26155.